src/anniversary/sf33rd/AcrSDK/ps2/flps2debug.c

```diff
@@ -0,0 +1,177 @@
+#include "sf33rd/AcrSDK/ps2/flps2debug.h"
+#include "common.h"
+#include "structs.h"
+
+#if defined(TARGET_PS2)
+#include "mw_stdarg.h"
+#else
+#include <stdarg.h>
+#endif
+
+void flPS2DebugStrClear();
+
+s32 flSetDebugMode(u32 flag) {
+    flDebugFlag = flag;
+    return 1;
+}
+
+void flGetSystemTime(void) {
+    f32 sys_time;
+
+    sys_time = ((u32)(*(s32 *)0x10000000));
+    sys_time = ((1000.0f * sys_time) / 60.0f) / 240.0f;
+    fptodp(sys_time);
+}
+
+void flPS2DebugInit(void) {
+    s32 var_s0;
+    s32 len;
+
+    flSetDebugMode(3);
+    flDebugSysMemHandleNum = 0;
+    flDebugSysMemEtc = 0;
+    flDebugSysMemTexture = 0;
+    flDebugSysMemClay = 0;
+    flDebugSysMemMotion = 0;
+
+    for (var_s0 = 0; var_s0 < 0x1000; var_s0++) {
+        flDebugSysMem[var_s0] = 0;
+    }
+
+    len = 0xE100;
+    flDebugStrHan = flPS2GetSystemMemoryHandle(len, 1);
+    flhDebugStr = flCreateTextureFromFile("IOP\\FONT8_8.TM2", 4);
+
+    if (flhDebugStr == 0) {
+        while (1) {}
+    }
+
+    flPrintColor(-1);
+    flPS2DebugStrClear();
+}
+
+void flPS2DebugStrClear(void) {
+    flDebugStrX = 0;
+    flDebugStrY = 0;
+    flDebugStrCtr = 0;
+}
+
+INCLUDE_ASM("asm/anniversary/nonmatchings/sf33rd/AcrSDK/ps2/flps2debug", flPS2DebugStrDisp);
+
+INCLUDE_ASM("asm/anniversary/nonmatchings/sf33rd/AcrSDK/ps2/flps2debug", flPrintL);
+
+s32 flPrintColor(u32 color) {
+    u8 a;
+    u8 r;
+    u8 g;
+    u8 b;
+
+    r = (color >> 16) & 0xFF;
+    g = (color >> 8) & 0xFF;
+    b = (color) & 0xFF;
+    a = (color >> 24) & 0xFF;
+
+    r >>= 1;
+    g >>= 1;
+    b >>= 1;
+
+    if (a == 0xFF) {
+        a = 0x80;
+    } else {
+        a >>= 1;
+    }
+
+    flDebugStrCol = (a << 24) | (r << 16) | (g << 8) | b;
+    return 1;
+}
+
+INCLUDE_ASM("asm/anniversary/nonmatchings/sf33rd/AcrSDK/ps2/flps2debug", flPS2DispSystemInfo);
+
+INCLUDE_ASM("asm/anniversary/nonmatchings/sf33rd/AcrSDK/ps2/flps2debug", flPS2DrawProbar);
+
+void flPS2LoadCheckFlush(void) {
+    s32 var_s0;
+    flLoadCheckCtr = 0;
+    flLoadCheckTimeOld = 0;
+
+    for (var_s0 = 0; var_s0 < 0x14; var_s0++) {
+        flLoadCheckTime[var_s0] = 0;
+    }
+}
+
+void flPS2SystemError(s32 error_level, s8 *format, ...) {
+    s8 str[512];
+    s32 len;
+    va_list args;
+
+    flFlip(0);
+    va_start(args, format);
+    vsprintf(str, format, args);
+    len = strlen(str);
+
+    while (1) {
+        flPrintL(0xA, 0x14, "%s", str);
```

Comment:
The first two args are x and y coordinates. It's easier to think about those in decimal.

```suggestion
        flPrintL(10, 20, "%s", str);
```